### ISSUE TITLE
Bug: CTCLoss can not run with cuda?

### ISSUE THREAD

**Issue by williamlzw** [dotnet/TorchSharp#815]
public interface LossInterface
    {
        public torch.Tensor Call(torch.Tensor pred, torch.Tensor label, torch.Tensor labelLength);
    }
    public class CtcLoss : LossInterface
    {
        private static CTCLoss _lossFunc;
        private static torch.Device _device;
        public CtcLoss(torch.Device device, int blank = 0, torch.nn.Reduction reduction = torch.nn.Reduction.Sum)
        {
            _lossFunc = torch.nn.CTCLoss(blank, reduction: reduction, zero_infinity:true);
            _lossFunc = _lossFunc.to(device);
            _device = device;
        }

        public torch.Tensor Call(torch.Tensor pred, torch.Tensor label, torch.Tensor labelLength)
        {
            pred = pred.permute(1, 0, 2);
            var batchSize = pred.size()[1];
            pred = torch.nn.functional.log_softmax(pred, 2);
            var value = pred.size()[0];
            var array = Enumerable.Repeat(value, (int)batchSize).ToArray();
            var predsLength = torch.from_array(array).to(_device);
            Console.WriteLine(pred.device_type);
            Console.WriteLine(label.device_type);
            Console.WriteLine(predsLength.device_type);
            Console.WriteLine(labelLength.device_type);
            return _lossFunc.forward(pred, label, predsLength, labelLength); //---------error
        }
    }

--------------------------------------------------------------------
System.Runtime.InteropServices.ExternalException:“Expected tensor to have CPU Backend, but got tensor with CUDA Backend (while checking arguments for cudnn_ctc_loss)
Exception raised from checkBackend at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\TensorUtils.cpp:222 (most recent call first):

**Comment by NiklasGustafsson** [comment]
@williamlzw -- could you put together a small repro case with (random) tensors of the right shape, etc.? Thanks!

**Comment by NiklasGustafsson** [comment]
Okay, so I ran the following code:

```C#
            int T = 50, C = 20, N = 16, S = 30, S_min = 10;

            var input = torch.nn.functional.log_softmax(torch.randn(T, N, C), 2).detach().requires_grad_().to(DeviceType.CUDA);
            var target = torch.randint(low: 1, high: C, size: (N, S), dtype: torch.@long).to(DeviceType.CUDA);
            var input_lengths = torch.full(size: N, value: T, dtype: torch.@long).to(DeviceType.CUDA);
            var target_lengths = torch.randint(low: S_min, high: S, size: N, dtype: torch.@long).to(DeviceType.CUDA);

            var ctc_loss = nn.CTCLoss().to(DeviceType.CUDA);
            var loss = ctc_loss.forward(input, target, input_lengths, target_lengths);
            loss.backward();
```

Without any exceptions.

**Comment by williamlzw** [comment]
It can run in cuda today
